Jenkinsfile

```diff
@@ -51,115 +140,101 @@ def gitMerge(String primaryBranch) {
+      resourceRequestCpu: '2',
+      resourceRequestMemory: '4Gi',
+      resourceLimitCpu: '7.5',
+      resourceLimitMemory: '4Gi'
+    ],
+    build: cacheName
+  ) {
+    props = readProperties(file: 'build.properties')
     String primaryBranch = props.primaryBranch
     assert primaryBranch != null && primaryBranch != ""
-    def runAllTests = env.BRANCH_NAME == primaryBranch
 
-    stage("Merge") {
+    stage('Merge') {
       // cancel build if primary branch doesn't merge cleanly
       gitMerge(primaryBranch)
     }
 
-    def platforms = [
-      [ os: "windows", browser: "chrome" ],
-      [ os: "windows", browser: "firefox" ],
-      [ os: "windows", browser: "MicrosoftEdge" ],
-      [ os: "macos", browser: "safari" ],
-      [ os: "macos", browser: "chrome" ],
-      [ os: "macos", browser: "firefox" ]
-    ]
-
-    def cleanAndInstall = {
-      echo "Installing tools"
-      exec("git clean -fdx modules scratch js dist")
+    stage('Install') {
       yarnInstall()
     }
 
-    def processes = [:]
-
-    // Browser tests
-    for (int i = 0; i < platforms.size(); i++) {
-      def platform = platforms.get(i)
-
-      def buckets = platform.buckets ?: 1
-      for (int bucket = 1; bucket <= buckets; bucket++) {
-        def suffix = buckets == 1 ? "" : "-" + bucket
-
-        // closure variable - don't inline
-        def c_bucket = bucket
-
-        def name = "${platform.os}-${platform.browser}${suffix}"
-
-        processes[name] = {
-          stage(name) {
-            node("bedrock-${platform.os}") {
-              echo("Bedrock tests for ${name}")
-
-              echo("Checking out code on build node: $NODE_NAME")
-              checkout(scm)
-
-              // windows tends to not have username or email set
-              tinyGit.addAuthorConfig()
-              gitMerge(primaryBranch)
-
-              cleanAndInstall()
-              exec("yarn ci")
-
-              echo("Running browser tests")
-              runBrowserTests(name, platform.browser, platform.os, c_bucket, buckets, runAllTests)
-            }
-          }
-        }
-      }
-    }
-
-    processes["headless-and-archive"] = {
-      stage("headless tests") {
-        runHeadlessTests(runAllTests)
-      }
-
-      if (env.BRANCH_NAME != primaryBranch) {
-        stage("Archive Build") {
-          exec("yarn tinymce-grunt prodBuild symlink:js")
-          archiveArtifacts artifacts: 'js/**', onlyIfSuccessful: true
-        }
-      }
-    }
-
-    stage("Install tools") {
-      cleanAndInstall()
-    }
-
     stage("Validate changelog") {
       // we use a changelog to run changie
       exec("yarn changie-merge")
     }
 
-    stage("Type check") {
+    stage('Type check') {
       withEnv(["NODE_OPTIONS=--max-old-space-size=1936"]) {
         exec("yarn ci-all-seq")
       }
     }
 
-    stage("Moxiedoc check") {
-      exec("yarn tinymce-grunt shell:moxiedoc")
+    stage('Moxiedoc check') {
+      sh 'yarn tinymce-grunt shell:moxiedoc'
```

Comment:
Should this be shell. I guess we will never really run moxidedoc on windows build nodes but seems inconsistent?

Reply:
Since they are in container everything should be fine running as `sh` however, for consistency I'm happy to keep everything as `exec`